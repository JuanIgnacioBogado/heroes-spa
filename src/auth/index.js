// Pages
export * from './pages';

// Context
export * from './context';

// Types
export * from './types/types';
