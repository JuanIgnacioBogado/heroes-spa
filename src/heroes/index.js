// Pages
export * from './pages';

// Route
export * from './routes/HeroesRoutes';
