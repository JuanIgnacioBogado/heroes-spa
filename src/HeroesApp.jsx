import { AppRouter } from './router/AppRouter';
import { AuthProvider } from './auth';

export const HeroesApp = () => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  );
};
