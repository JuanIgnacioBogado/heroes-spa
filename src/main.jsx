import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';

import { HeroesApp } from './HeroesApp';

import './index.css';

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </StrictMode>
);
