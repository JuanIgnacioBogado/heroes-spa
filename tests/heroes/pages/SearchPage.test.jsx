import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { SearchPage } from '../../../src/heroes';

const mockedSetSearchParams = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useSearchParams: () => [
    jest.requireActual('react-router-dom').useSearchParams()[0],
    mockedSetSearchParams
  ]
}));

describe('SearchPage', () => {
  beforeEach(() => jest.clearAllMocks());

  test('should to show correctly whit default values', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('should to show Batman and input whit the queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByPlaceholderText(/search a hero/i);
    expect(input.value).toBe('batman');

    const h5 = screen.getByRole('heading', { name: /batman/i });
    expect(h5).toBeTruthy();

    const img = screen.getByAltText(/batman/i);
    expect(img.src).toContain('/heroes/dc-batman.jpg');
  });

  test('should to show an error if hero not found', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=calamardo']}>
        <SearchPage />
      </MemoryRouter>
    );

    const alertError = screen.getByText(/no hero with/i);
    expect(alertError).toBeTruthy();
  });

  test('should to call setSearchParams', () => {
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByPlaceholderText(/search a hero/i);
    fireEvent.input(input, { target: { value: 'superman' } });
    fireEvent.submit(input);

    expect(mockedSetSearchParams).toBeCalledWith({ q: 'superman' });
  });
});
