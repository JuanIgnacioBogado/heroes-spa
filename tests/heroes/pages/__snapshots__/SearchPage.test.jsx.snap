// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should to show correctly whit default values 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
